### PR TITLE
Switch lxml type stubs library

### PR DESCRIPTION
**Pull request by eliotwrobson**
See title, the old one is no longer considered up to date.

### CODE REVIEW

**Comment by jonatanschroeder** [comment]
Want to incorporate the changes in #9355?

**Comment by eliotwrobson** [comment]
> Want to incorporate the changes in #9355?

yes! feel free to commit to this branch directly @jonatanschroeder

File: apps/prairielearn/python/traverse.py
Comment (eliotwrobson) [inline apps/prairielearn/python/traverse.py:113]: @nwalters512 not sure if this is going to cause issues: https://developer.mozilla.org/en-US/docs/Web/API/ProcessingInstruction

**Review by SethPoulsen — APPROVED** [review]
Seems fine to me.

File: apps/prairielearn/python/traverse.py
Comment (nwalters512) [reply apps/prairielearn/python/traverse.py:113]: I'm confused about what's happening here. Why is this annotation necessary? Locally when I run `make deps && make typecheck-python` I get `Unnecessary "# type: ignore" comment` for this line.

Comment (eliotwrobson) [reply apps/prairielearn/python/traverse.py:113]: If you have the old stubs library installed, you won't see an error. But if you uninstall that one and install the new one, that unnecessary comment error should go away. See this comment: https://github.com/abelcheung/types-lxml/blob/main/lxml-stubs/html/_element.pyi#L228

Comment (nwalters512) [reply apps/prairielearn/python/traverse.py:113]: Sounds like they made a horrible assumption. I would strongly encourage you to file an issue or PR upstream.

Comment (eliotwrobson) [reply apps/prairielearn/python/traverse.py:113]: I can file an issue. Just wanted to make sure that this wasn't a bug on our end. This is kinda unrelated to the point of this PR, and we have fewer ignore comments overall, so going to go ahead and merge.

Comment (nwalters512) [reply apps/prairielearn/python/traverse.py:113]: It's definitely not a bug, `lxml` very much supports it:

https://github.com/lxml/lxml/blob/3c902168c46c2e55b69f38b1b56d57c5296958f6/src/lxml/html/__init__.py#L673

https://github.com/lxml/lxml/blob/3c902168c46c2e55b69f38b1b56d57c5296958f6/src/lxml/html/__init__.py#L716

We have tests that specifically rely on this:

https://github.com/PrairieLearn/PrairieLearn/blob/136e3ed3a6825b8c56ec26c3c5bcf2db65d4ccaa/apps/prairielearn/python/traverse_test.py#L305-L310

Before merging, please file the issue and include a link to it inline so that future maintainers can see why the `# type: ignore` was added and check if the upstream issue was fixed yet. I make a habit of doing this for broken JS types/implementations and it's proven useful.

Comment (eliotwrobson) [reply apps/prairielearn/python/traverse.py:113]: Ah yeah, if our tests specifically rely on getting that class then it's probably just a mistake in the types.

Comment (eliotwrobson) [reply apps/prairielearn/python/traverse.py:117]: Issue filed and comment added here: https://github.com/PrairieLearn/PrairieLearn/pull/9356/commits/98a1c02020b13a2ce4d4d5a0c1f4fe5542ccd5a0

Comment (nwalters512) [reply apps/prairielearn/python/traverse.py:117]: Thank you!

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**
